### ISSUE TITLE
Reading LAS file from http(s) URIs

### ISSUE THREAD

**Issue by gsakkis** [PDAL/PDAL#3362]
When running PDAL installed via Conda I can access remote http(s) LAS files:
```
$ pdal --version
--------------------------------------------------------------------------------------------------------------------------------------------------------------------
pdal 2.2.0 (git-version: 65c8de)
--------------------------------------------------------------------------------------------------------------------------------------------------------------------

$ pdal info --summary http://data.entwine.io/red-rocks.laz | jq keys
[
  "filename",
  "now",
  "pdal_version",
  "reader",
  "summary"
]
```

When building from source it tries to parse it as a local file:
```
$ pdal info --summary https://data.entwine.io/red-rocks.laz
PDAL: pdalboost::filesystem::file_size: No such file or directory: "https://data.entwine.io/red-rocks.laz"
```

So I'm guessing it's missing a library or a build option but looking at the dependencies I don't see anything obvious (e.g. `curl` is available).

**Comment by hobu** [comment]
This works for me on OSX conda. 

I believe PDAL tries to write to `/tmp` as its working location when fetching remote data. Could it be that it can't write there on your system?

**Comment by abellgithub** [comment]
Can you please provide the output from cmake configuration?

**Comment by gsakkis** [comment]
> I believe PDAL tries to write to /tmp as its working location when fetching remote data. Could it be that it can't write there on your system?

I checked, /tmp is writable by all.

cmake output:
```
-- Could NOT find LIBEXECINFO (missing: LIBEXECINFO_LIBRARY) 
-- Could NOT find LIBUNWIND (missing: LIBUNWIND_LIBRARY LIBUNWIND_INCLUDE_DIR) 
-- The following features have been enabled:

 * TileDB plugin, read/write data from TileDB
 * Unit tests, PDAL unit tests

-- The following OPTIONAL packages have been found:

 * ZSTD
   General compression support
 * LibXml2
 * PkgConfig
 * OpenSSL (required version >= 1.1)
 * PythonInterp
 * absl

-- The following RECOMMENDED packages have been found:

 * LASzip (required version >= 3.1)
   Provides LASzip compression
   Provides LASzip compression

-- The following REQUIRED packages have been found:

 * GDAL (required version >= 2.2.0)
   Provides general purpose raster, vector, and reference system support
 * GeoTIFF
 * ZLIB
   Compression support in BPF
 * CURL
 * TileDB (required version >= 1.4.1), <https://www.tiledb.io>
   TileDB support
 * Threads, The thread library of the system

-- The following features have been disabled:

 * Bash completion, completion for PDAL command line
 * CPD plugin, Coherent Point Drift (CPD) computes rigid or nonrigid transformations between point sets
 * I3S plugin, Read from a I3S server or from a SLPK file
 * Icebridge plugin, read data in the Icebridge format
 * HDF plugin, read data in the HDF format
 * Matlab plugin, write data to a .mat file
 * MrSID plugin, read data in the MrSID format
 * NITF plugin, read/write LAS data wrapped in NITF
 * OpenSceneGraph plugin, read/write OpenSceneGraph objects
 * Oracle OCI plugin, Read/write point clould patches to Oracle
 * PostgreSQL PointCloud plugin, read/write PostgreSQL PointCloud objects
 * RiVLib plugin, read data in the RXP format
 * rdblib plugin, read data in the RDB format
 * MBIO plugin, add features that depend on MBIO
 * FBX plugin, add features that depend on FBX
 * TEASER++ plugin, TEASER++ computes transformations between point sets
 * E57 plugin, read/write data to and from e57 format

-- The following OPTIONAL packages have not been found:

 * PostgreSQL
 * Libexecinfo
 * Libunwind

-- Configuring done
-- Generating done
```

**Comment by abellgithub** [comment]
This was broken by #3226. I'll try to fix soon

**Comment by abellgithub** [comment]
Fixed by #3363
